tests/pkg-layering/main.yml

```diff
@@ -0,0 +1,319 @@
+
+    - name: Set packages when deployment 1 is not booted
+      set_fact:
+        pending_pkgs: "{{ pending_json['deployments'][1]['packages'] }}"
+      when: pending_json['deployments'][1] is defined and not pending_json['deployments'][1]['booted']
+
+    - name: Fail if {{ g_pkg2 }} is in rpm-ostree status output for pending packages
+      fail:
+        msg: "{{ g_pkg2 }} in rpm-ostree status output"
+      when: "'{{ g_pkg2 }}' in pending_pkgs"
+
+    - name: Fail if {{ g_pkg1 }} is not in rpm-ostree status output for pending packages
+      fail:
+        msg: "{{ g_pkg1 }} not in rpm-ostree status output"
+      when: "'{{ g_pkg1 }}' not in pending_pkgs"
+
+    - name: Fail if {{ g_pkg1 }} is installed
+      command: command -v {{ g_pkg1 }}
+      register: binary
+      failed_when: binary.rc != 1
+
+    # Verify that after additional repos are enabled (i.e. EPEL), rpm-ostree install|uninstall can be successfully used for packages in those repos.
+    - name: Retrieve EPEL rpm
+      get_url:
+        url: https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+        dest: /tmp
+        mode: 0600
+
+    - name: Extract GPG Key to install directory
+      shell: >
+        rpm2cpio /tmp/epel-release-latest-7.noarch.rpm |
+        cpio -iv --to-stdout ./etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7 >
+        /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
+
+    - name: Extract EPEL repo file to install directory
+      shell: >
+        rpm2cpio /tmp/epel-release-latest-7.noarch.rpm |
+        cpio -iv --to-stdout ./etc/yum.repos.d/epel.repo >
+        /etc/yum.repos.d/epel.repo
+
+    - name: Install {{ g_epel_pkg }} from EPEL
+      command: rpm-ostree install {{ g_epel_pkg }} -r
+      async: 60
+      poll: 0
+      ignore_errors: true
+
+    - name: wait for host to go down
+      local_action: 
+        wait_for host={{ inventory_hostname }}
+        port=22
+        state=absent
+        delay=1
+        timeout=120
+      become: false
+
+    - name: wait for hosts to come back up (inventory_hostname)
+      local_action: 
+        wait_for host={{ inventory_hostname }}
+        port=22
+        state=started
+        delay=30
+        timeout=120
+      become: false
+
+    - name: Get rpm-ostree status --json output
+      command: rpm-ostree status --json
+      register: installed
+
+    - name: Convert rpm-ostree status output to jinja2 json
+      set_fact:
+        installed_json: "{{ installed.stdout | from_json }}"
+
+    - name: Set packages when deployment 0 is booted
+      set_fact:
+        installed_pkgs: "{{ installed_json['deployments'][0]['packages'] }}"
+      when: installed_json['deployments'][0] is defined and installed_json['deployments'][0]['booted']
+
+    - name: Set packages when deployment 1 is booted
+      set_fact:
+        installed_pkgs: "{{ installed_json['deployments'][1]['packages'] }}"
+      when: installed_json['deployments'][1] is defined and installed_json['deployments'][1]['booted']
+
```

Comment:
I moved these into an include.  It is basically a duplication of the verification roles.